### PR TITLE
Upgrade to Qt 6

### LINKED ISSUE #438: Update to Qt 6.2
Enhancement Request
--------------
Update Qt to 6.2 LTS version to support macOS on Apple Silicon

### PR DESCRIPTION
**Pull request by macumber**
Fixes #438

### CODE REVIEW

File: src/openstudio_app/CMakeLists.txt [hunk 1]
```diff
@@ -413,17 +382,21 @@ if( APPLE )
...
+if(APPLE)
+  # This does some "magic" **including on Linux** (cf #345) (On linux it will install in ./bin/ the libopenstudio.so, and libicu* stuff)
+  install(CODE "
+    # APPS might be a symbolic link which does not work with fixup_bundle
+    # expand APPS to a real path
+    get_filename_component(APPS_EXPANDED ${APPS} REALPATH)
+    include(BundleUtilities)
+    # This file defines some libs we are excluding. Some like concrt140.dll are installed already via InstallRequiredSystemLibraries
+    include(\"${CMAKE_SOURCE_DIR}/CMake/FixupBundleOverrides.cmake\")
+    fixup_bundle(\"${APPS_EXPANDED}\" \"${QTPLUGINS}\" \"${DIRS}\")
```
Comment (jmarrec) [inline src/openstudio_app/CMakeLists.txt:396]: This is what it used to be!

```suggestion
    fixup_bundle(\"\${APPS_EXPANDED}\" \"${QTPLUGINS}\" \"${DIRS}\")
```

**Review by jmarrec — CHANGES_REQUESTED** [review]

File: .github/workflows/app_build.yml [hunk 2]
```diff
@@ -28,7 +28,7 @@ env:
   CPACK_SOURCE_TXZ: OFF
   CPACK_SOURCE_TZ: OFF
   CPACK_SOURCE_ZIP: OFF
-  QT_VERSION: 5.15.0
+  QT_VERSION: 6.2.1
```
Comment (jmarrec) [inline .github/workflows/app_build.yml:31]: - [ ] TODO

```suggestion
  QT_VERSION: 6.2.2
```

**Review by jmarrec — CHANGES_REQUESTED** [review]

File: FindOpenStudioSDK.cmake [hunk 3]
```diff
@@ -17,7 +17,7 @@ if(APPLE)
     set(OPENSTUDIO_EXPECTED_HASH f215e3db09f3914cf0f2040f601147e1)
     set(OPENSTUDIO_PLATFORM "Darwin-arm64")
   else()
-    set(OPENSTUDIO_EXPECTED_HASH 5c81710a1a8fd0cb8f461345bbfb5a25)
+    set(OPENSTUDIO_EXPECTED_HASH 77c2bf77f07dc2e2af7658df55940c49)
```
Comment (macumber) [inline FindOpenStudioSDK.cmake:20]: Thanks @jmarrec !

**Comment by macumber** [comment]
@jmarrec I was hoping to get this in in the next week if things look good, is that ok with you?

**Comment by jmarrec** [comment]
@macumber Sure. But do note that it means dropping 18.04 support, as Qt 6 doesn't support it.

**Comment by macumber** [comment]
@jmarrec I am good to merge this now.  Do you want to do any final review or testing on Ubuntu 20.04?

File: .github/workflows/app_build.yml [hunk 4]
```diff
@@ -28,36 +28,22 @@ env:
   CPACK_SOURCE_TXZ: OFF
   CPACK_SOURCE_TZ: OFF
   CPACK_SOURCE_ZIP: OFF
-  QT_VERSION: 5.15.0
+  QT_VERSION: 6.2.3
```
[the remark this answers is not in the document]
Comment (jmarrec) [reply .github/workflows/app_build.yml]: Qt 6.3.0 is planned in 4 days (original: 16.03.2022 , updated: 12.04.2022)

**Review by jmarrec — APPROVED** [review]
Assuming CI comes clean, we'll merge this one. We can bump to Qt 6.3.0 when it comes out.

**Pull request merged**